### PR TITLE
Improve the createNonzero test

### PR DESCRIPTION
**Pull request by lrettig**
See commit comment for details.

### CODE REVIEW

**Comment by lrettig** [comment]
Over to you @hugo-dc :)

File: src/GeneralStateTestsFiller/stEWASMTests/createNonzeroFiller.yml [hunk 1]
```diff
@@ -23,37 +52,39 @@ createNonzero:
...
-    #         (i32.store (get_local $memPointerReturnData) (i32.const 0xabcd))
+    #         (i32.store (get_local $memPointerStorageKey4) (i32.const 3))
 
     #         ;; get call value
     #         ;; result gets stored in pointer loc
     #         (call $getCallValue (get_local $memPointerCallValue))
 
     #         ;; write it to storage
-    #         (call $storageStore (get_local $memPointerStorageKey3) (get_local $memPointerCallValue))
+    #         (call $storageStore (get_local $memPointerStorageKey4) (get_local $memPointerCallValue))
```
Comment (hugo-dc) [inline src/GeneralStateTestsFiller/stEWASMTests/createNonzeroFiller.yml:79]: Do we still need to call $getCallValue and store the result of  here?

**Review by hugo-dc — COMMENTED** [review]
Everything looks good, I got confused on why the wasm code which return the actual contract code is storing callValue

File: src/GeneralStateTestsFiller/stEWASMTests/createNonzeroFiller.yml (hunk 1 above)
Comment (lrettig) [reply src/GeneralStateTestsFiller/stEWASMTests/createNonzeroFiller.yml:79]: Why not? It's checked in the expect section on line 207. Strictly speaking value transfer can be checked by expecting a particular balance, but I don't see any harm in this check.

Comment (hugo-dc) [reply src/GeneralStateTestsFiller/stEWASMTests/createNonzeroFiller.yml:79]: OK I see it now, thanks Lane, no more comments

**Review by hugo-dc — APPROVED** [review]

**Pull request merged**